docs/docs/tutorial/creating-a-source-plugin/part-5/index.mdx

```diff
@@ -0,0 +1,329 @@
+        getNodes().forEach((node) => {
+          if (node.internal.owner !== `plugin`) {
+            return
+          }
+
+          touchNode(node)
+        })
+        // highlight-end
+
+        isFirstSource = false
+      }
+
+      // Rest of sourceNodes
+    }
+
+    // Rest of file
+    ```
+
+    `plugin` is the name of your source plugin for this tutorial. Adjust it to the actual name once you write your actual source plugin.
+
+With the code complete, your source plugin won't discard GraphQL nodes that should still exist on subsequent runs.
+
+### 2. Create and use sync token
+
+The details of this step completely depend on the remote API you're working with. Visit the API's documentation to check if it supports delta updates and how you can use them.
+
+Independent from what sync token you need to save (be it a timestamp, a token, or something else), you've already learned in [Part 4](/docs/tutorial/creating-a-source-plugin/part-4/#cache-api) how you can use the `cache` API to save and use that token.
+
+Use that sync token on subsequent calls to the API to only receive delta updates.
+
+<Collapsible
+  summary={<em>Example: <code>gatsby-source-contentful</code></em>}
+>
+
+You can view the whole source code under the GitHub permalinks for [source-nodes](https://github.com/gatsbyjs/gatsby/blob/240f5a9dc9f0cc5f5369e3087244378aad3d5ccf/packages/gatsby-source-contentful/src/source-nodes.js#L87-L144) and [fetch](https://github.com/gatsbyjs/gatsby/blob/2794ed31f693c74921d20011aa4b060bc32eaec0/packages/gatsby-source-contentful/src/fetch.js#L236).
+
+1. First, a unique `sourceId` string is generated from the `spaceId` and `environment` which are both set in the plugin's options.
+
+    ```js
+    const sourceId = `${pluginConfig.get(`spaceId`)}-${pluginConfig.get(
+      `environment`
+    )}`
+    ```
+
+1. `sourceId` used for the `CACHE_SYNC_TOKEN` string:
+
+    ```js
+    const CACHE_SYNC_TOKEN = `contentful-sync-token-${sourceId}`
+    ```
+
+1. The next part might be unfamiliar to you since the tutorial hasn't mentioned it so far:
+
+    ```js
+    const syncToken =
+      store.getState().status.plugins?.[`gatsby-source-contentful`]?.[
+        CACHE_SYNC_TOKEN
+      ]
+    ```
+
+    In [Part 4](/docs/tutorial/creating-a-source-plugin/part-4/#cache-api) you've learned how to use the `cache` API to save a sync token. You can also use the [`setPluginStatus` API](/docs/reference/config-files/actions/#setPluginStatus) to achieve the same end result. While `gatsby-source-contentful` uses `setPluginStatus`, we'd recommend using the `cache` API instead due to it's simpler usage.
+
+    Regardless, in the end a `syncToken` is saved inside a variable.
+
+1. The `syncToken` is passed to the `fetchContent` utility function:
+
+    ```js
+    const {
+      currentSyncData,
+      // + other data
+    } = await fetchContent({ syncToken, pluginConfig, reporter })
+    ```
+
+    And coming back is the `currentSyncData` or: The delta update.
```

Comment:
This line was confusing, is `currentSyncData` the delta update? Or is it supposed to mean either the `currentSyncData` or the delta update come back?

First option:
```suggestion
    And coming back is the `currentSyncData`, the delta update.
```

Second option:
```suggestion
    And coming back is the `currentSyncData`, or The delta update.
```